src/lib/mongoBackend/mongoRegistrationGet.cpp

```diff
@@ -638,13 +639,52 @@ bool mongoLdRegistrationsGet
     }
   }
 
+  if (ciP->uriParam["type"] != "")
+  {
+    char*                       typeList = (char*) ciP->uriParam["type"].c_str();
+    std::vector<std::string>    typeVec;
+    int                         types;
+    mongo::BSONObjBuilder       bsonInExpression;
+    mongo::BSONArrayBuilder     bsonArray;
+    char*                       details;
+    char                        typeExpanded[256];
+
+    types = stringSplit(typeList, ',', typeVec);
+
+    if (types > 0)
+    {
+      for (int ix = 0; ix < types; ix++){
+        char type[typeVec[ix].size() + 1];
```

Comment:
Why do you need this variable?
Should be enough with `typeExpanded`

Reply:
Cause the str functions need a param of char type, But your read your tip about c_str(). Thanks again...

Reply:
Fixed in 818e634